src/m365/pp/commands/card/card-remove.spec.ts

```diff
@@ -0,0 +1,223 @@
+      if (command === PpCardGetCommand) {
+        return ({
+          stdout: `{ "overwritetime": "1900-01-01T00:00:00Z", "_owningbusinessunit_value": "b419f090-fe22-ec11-b6e5-000d3ab596a1", "solutionid": "fd140aae-4df4-11dd-bd17-0019b9312238", "componentidunique": "e2b1d019-bd9a-491a-b888-693740711319", "_owninguser_value": "4f175d04-b952-ed11-bba2-000d3adf774e", "statecode": 0, "statuscode": 1, "ismanaged": false, "cardid": "${validId}", "_ownerid_value": "4f175d04-b952-ed11-bba2-000d3adf774e", "componentstate": 0, "modifiedon": "2022-10-29T08:22:46Z", "name": "${validName}", "_modifiedby_value": "4f175d04-b952-ed11-bba2-000d3adf774e", "versionnumber": 4463945, "createdon": "2022-10-29T08:22:46Z", "description": " ", "_createdby_value": "4f175d04-b952-ed11-bba2-000d3adf774e", "overriddencreatedon": null, "schemaversion": null, "importsequencenumber": null, "tags": null, "_modifiedonbehalfby_value": null, "utcconversiontimezonecode": null, "publishdate": null, "_createdonbehalfby_value": null, "hiddentags": null, "remixsourceid": null, "sizes": null, "coowners": null, "_owningteam_value": null, "publishsourceid": null, "timezoneruleversionnumber": null, "iscustomizable": { "Value": true, "CanBeChanged": true, "ManagedPropertyLogicalName": "iscustomizableanddeletable"}}`
+        });
+      }
+
+      throw new CommandError('Unknown case');
+    });
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.1/cards(${validId})`) {
+        return { statusCode: 200 };
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+    await command.action(logger, {
+      options: {
+        debug: true,
+        environment: 'Default-eff8592e-e14a-4ae8-8771-d96d5c549e1c',
+        name: 'CLI 365 Card'
+      }
+    });
+    assert(loggerLogToStderrSpy.called);
+  });
+
+  it('removes the specified card owned by the currently signed-in user without prompt for confirm', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
+
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.1/cards(${validId})`) {
+        return { statusCode: 200 };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        environment: validEnvironment,
+        id: validId,
+        confirm: true
+      }
+    });
+    assert(loggerLogToStderrSpy.called);
+  });
+
+  it('correctly handles API OData error', async () => {
+    const errorMessage = `The environment '${validEnvironment}' could not be retrieved. See the inner exception for more details: undefined`;
+    sinon.stub(request, 'get').callsFake(async () => { throw errorMessage; });
```

Comment:
Instead of simulating an environment error, let's simulate an error of the `remove` request.
This will return an error that looks like this:

```
"error": {
    "error": {
      "message": "Something went wrong"
    }
  }
```